--- a/decode.go
+++ b/decode.go
@@ -668,7 +668,7 @@
 }
 
 func (d *Decoder) deleteStructKeys(structType reflect.Type, unknownFields map[string]ast.Node) error {
-	if structType.Kind() == reflect.Ptr {
+	if structType.Kind() == reflect.Pointer {
 		structType = structType.Elem()
 	}
 	structFieldMap, err := structFieldMap(structType)
@@ -802,7 +802,7 @@
 	if unmarshaler, ok := iface.(InterfaceUnmarshalerContext); ok {
 		if err := unmarshaler.UnmarshalYAML(ctx, func(v interface{}) error {
 			rv := reflect.ValueOf(v)
-			if rv.Type().Kind() != reflect.Ptr {
+			if rv.Type().Kind() != reflect.Pointer {
 				return ErrDecodeRequiredPointerType
 			}
 			if err := d.decodeValue(ctx, rv.Elem(), src); err != nil {
@@ -818,7 +818,7 @@
 	if unmarshaler, ok := iface.(InterfaceUnmarshaler); ok {
 		if err := unmarshaler.UnmarshalYAML(func(v interface{}) error {
 			rv := reflect.ValueOf(v)
-			if rv.Type().Kind() != reflect.Ptr {
+			if rv.Type().Kind() != reflect.Pointer {
 				return ErrDecodeRequiredPointerType
 			}
 			if err := d.decodeValue(ctx, rv.Elem(), src); err != nil {
@@ -917,7 +917,7 @@
 	}
 	valueType := dst.Type()
 	switch valueType.Kind() {
-	case reflect.Ptr:
+	case reflect.Pointer:
 		if dst.IsNil() {
 			return nil
 		}
@@ -1051,7 +1051,7 @@
 
 func (d *Decoder) createDecodableValue(typ reflect.Type) reflect.Value {
 	for {
-		if typ.Kind() == reflect.Ptr {
+		if typ.Kind() == reflect.Pointer {
 			typ = typ.Elem()
 			continue
 		}
@@ -1061,7 +1061,7 @@
 }
 
 func (d *Decoder) castToAssignableValue(value reflect.Value, target reflect.Type, src ast.Node) (reflect.Value, error) {
-	if target.Kind() != reflect.Ptr {
+	if target.Kind() != reflect.Pointer {
 		if !value.Type().AssignableTo(target) {
 			return reflect.Value{}, errors.ErrTypeMismatch(target, value.Type(), src.GetToken())
 		}
@@ -1108,7 +1108,7 @@
 	} else {
 		newValue = d.createDecodableValue(typ)
 	}
-	for defaultVal.Kind() == reflect.Ptr {
+	for defaultVal.Kind() == reflect.Pointer {
 		defaultVal = defaultVal.Elem()
 	}
 	if defaultVal.IsValid() && defaultVal.Type().AssignableTo(newValue.Type()) {
@@ -1187,7 +1187,7 @@
 }
 
 func (d *Decoder) setDefaultValueIfConflicted(v reflect.Value, fieldMap StructFieldMap) error {
-	for v.Type().Kind() == reflect.Ptr {
+	for v.Type().Kind() == reflect.Pointer {
 		v = v.Elem()
 	}
 	typ := v.Type()
@@ -1321,7 +1321,7 @@
 	structType := dst.Type()
 	srcValue := reflect.ValueOf(src)
 	srcType := srcValue.Type()
-	if srcType.Kind() == reflect.Ptr {
+	if srcType.Kind() == reflect.Pointer {
 		srcType = srcType.Elem()
 		srcValue = srcValue.Elem()
 	}
@@ -1374,7 +1374,7 @@
 			if !fieldValue.CanSet() {
 				return fmt.Errorf("cannot set embedded type as unexported field %s.%s", field.PkgPath, field.Name)
 			}
-			if fieldValue.Type().Kind() == reflect.Ptr && src.Type() == ast.NullType {
+			if fieldValue.Type().Kind() == reflect.Pointer && src.Type() == ast.NullType {
 				// set nil value to pointer
 				fieldValue.Set(reflect.Zero(fieldValue.Type()))
 				continue
@@ -1421,7 +1421,7 @@
 		}
 		delete(unknownFields, structField.RenderName)
 		fieldValue := dst.FieldByName(field.Name)
-		if fieldValue.Type().Kind() == reflect.Ptr && src.Type() == ast.NullType {
+		if fieldValue.Type().Kind() == reflect.Pointer && src.Type() == ast.NullType {
 			// set nil value to pointer
 			fieldValue.Set(reflect.Zero(fieldValue.Type()))
 			continue
@@ -1552,7 +1552,7 @@
 	var foundErr error
 	for iter.Next() {
 		v := iter.Value()
-		if elemType.Kind() == reflect.Ptr && v.Type() == ast.NullType {
+		if elemType.Kind() == reflect.Pointer && v.Type() == ast.NullType {
 			// set nil value to pointer
 			arrayValue.Index(idx).Set(reflect.Zero(elemType))
 		} else {
@@ -1596,7 +1596,7 @@
 	var foundErr error
 	for iter.Next() {
 		v := iter.Value()
-		if elemType.Kind() == reflect.Ptr && v.Type() == ast.NullType {
+		if elemType.Kind() == reflect.Pointer && v.Type() == ast.NullType {
 			// set nil value to pointer
 			sliceValue = reflect.Append(sliceValue, reflect.Zero(elemType))
 			continue
@@ -1766,7 +1766,7 @@
 				return err
 			}
 		}
-		if valueType.Kind() == reflect.Ptr && value.Type() == ast.NullType {
+		if valueType.Kind() == reflect.Pointer && value.Type() == ast.NullType {
 			// set nil value to pointer
 			mapValue.SetMapIndex(k, reflect.Zero(valueType))
 			continue
@@ -1992,7 +1992,7 @@
 // and stores it in the value pointed to by v with context.Context.
 func (d *Decoder) DecodeContext(ctx context.Context, v interface{}) error {
 	rv := reflect.ValueOf(v)
-	if !rv.IsValid() || rv.Type().Kind() != reflect.Ptr {
+	if !rv.IsValid() || rv.Type().Kind() != reflect.Pointer {
 		return ErrDecodeRequiredPointerType
 	}
 	if d.isInitialized() {
@@ -2018,7 +2018,7 @@
 // DecodeFromNodeContext decodes node into the value pointed to by v with context.Context.
 func (d *Decoder) DecodeFromNodeContext(ctx context.Context, node ast.Node, v interface{}) error {
 	rv := reflect.ValueOf(v)
-	if rv.Type().Kind() != reflect.Ptr {
+	if rv.Type().Kind() != reflect.Pointer {
 		return ErrDecodeRequiredPointerType
 	}
 	if !d.isInitialized() {
--- a/encode.go
+++ b/encode.go
@@ -275,7 +275,7 @@
 		return true
 	}
 	kind := v.Type().Kind()
-	if kind == reflect.Ptr && v.IsNil() {
+	if kind == reflect.Pointer && v.IsNil() {
 		return true
 	}
 	if kind == reflect.Interface && v.IsNil() {
@@ -460,7 +460,7 @@
 		return e.encodeFloat(v.Float(), 32), nil
 	case reflect.Float64:
 		return e.encodeFloat(v.Float(), 64), nil
-	case reflect.Ptr:
+	case reflect.Pointer:
 		if value := e.encodePtrAnchor(v, column); value != nil {
 			return value, nil
 		}
@@ -738,7 +738,7 @@
 func (e *Encoder) isOmittedByOmitZero(v reflect.Value) bool {
 	kind := v.Kind()
 	if z, ok := v.Interface().(IsZeroer); ok {
-		if (kind == reflect.Ptr || kind == reflect.Interface) && v.IsNil() {
+		if (kind == reflect.Pointer || kind == reflect.Interface) && v.IsNil() {
 			return true
 		}
 		return z.IsZero()
@@ -746,7 +746,7 @@
 	switch kind {
 	case reflect.String:
 		return len(v.String()) == 0
-	case reflect.Interface, reflect.Ptr, reflect.Slice, reflect.Map:
+	case reflect.Interface, reflect.Pointer, reflect.Slice, reflect.Map:
 		return v.IsNil()
 	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
 		return v.Int() == 0
@@ -775,7 +775,7 @@
 	switch v.Kind() {
 	case reflect.String:
 		return len(v.String()) == 0
-	case reflect.Interface, reflect.Ptr:
+	case reflect.Interface, reflect.Pointer:
 		return v.IsNil()
 	case reflect.Slice, reflect.Map:
 		return v.Len() == 0
@@ -801,7 +801,7 @@
 func (e *Encoder) isOmittedByOmitEmptyTag(v reflect.Value) bool {
 	kind := v.Kind()
 	if z, ok := v.Interface().(IsZeroer); ok {
-		if (kind == reflect.Ptr || kind == reflect.Interface) && v.IsNil() {
+		if (kind == reflect.Pointer || kind == reflect.Interface) && v.IsNil() {
 			return true
 		}
 		return z.IsZero()
@@ -809,7 +809,7 @@
 	switch kind {
 	case reflect.String:
 		return len(v.String()) == 0
-	case reflect.Interface, reflect.Ptr:
+	case reflect.Interface, reflect.Pointer:
 		return v.IsNil()
 	case reflect.Slice, reflect.Map:
 		return v.Len() == 0
@@ -864,7 +864,7 @@
 			anchorName = snode.Value
 		}
 	}
-	if fieldValue.Kind() == reflect.Ptr {
+	if fieldValue.Kind() == reflect.Pointer {
 		e.setAnchor(fieldValue.Pointer(), anchorName)
 	}
 	return anchorNode, nil
@@ -987,7 +987,7 @@
 				anchorName = snode.Value
 			}
 		}
-		if inlineAnchorValue.Kind() == reflect.Ptr {
+		if inlineAnchorValue.Kind() == reflect.Pointer {
 			e.setAnchor(inlineAnchorValue.Pointer(), anchorName)
 		}
 		return anchorNode, nil
@@ -1001,7 +1001,7 @@
 	}
 
 	switch v.Type().Kind() {
-	case reflect.Ptr:
+	case reflect.Pointer:
 		return v.Pointer()
 	case reflect.Interface:
 		return e.toPointer(v.Elem())
--- a/parser/token.go
+++ b/parser/token.go
@@ -125,16 +125,16 @@
 func (t *Token) Dump() {
 	ctx := new(groupTokenRenderContext)
 	if t.Token != nil {
-		fmt.Fprint(os.Stdout, t.Token.Value)
+		_, _ = fmt.Fprint(os.Stdout, t.Token.Value)
 		return
 	}
 	t.Group.dump(ctx)
-	fmt.Fprintf(os.Stdout, "\n")
+	_, _ = fmt.Fprintf(os.Stdout, "\n")
 }
 
 func (t *Token) dump(ctx *groupTokenRenderContext) {
 	if t.Token != nil {
-		fmt.Fprint(os.Stdout, t.Token.Value)
+		_, _ = fmt.Fprint(os.Stdout, t.Token.Value)
 		return
 	}
 	t.Group.dump(ctx)
@@ -165,12 +165,12 @@
 
 func (g *TokenGroup) dump(ctx *groupTokenRenderContext) {
 	num := ctx.num
-	fmt.Fprint(os.Stdout, colorize(num, "("))
+	_, _ = fmt.Fprint(os.Stdout, colorize(num, "("))
 	ctx.num++
 	for _, tk := range g.Tokens {
 		tk.dump(ctx)
 	}
-	fmt.Fprint(os.Stdout, colorize(num, ")"))
+	_, _ = fmt.Fprint(os.Stdout, colorize(num, ")"))
 }
 
 func (g *TokenGroup) RawToken() *token.Token {
